dist/profile/templates/jenkinscontroller/casc/clouds.yaml.erb

```
@@ -479,6 +479,8 @@ jenkins:
           #!/bin/sh
           set -eux
 
+          systemctl stop sshd.service
```

why?

---

```suggestion
          # Ensure Jenkins controller does not start the agent in process in its workspace until we are ready
          systemctl stop sshd.service
```

> why?

Cloud Init might not be finished when Jenkins establish the SSH connection and start writing in the agent workspace. If it happens before mounting the NVMe, it will fail the agent.

---

by locking the sshd at the beginning and restarting it at the end, we ensure that the agent isn't seen as ready as long as the cloud-init is not done ... (it's already assumed by the loop within the initscript but this loop isn't working on windows) as we set the new mount point as the workspace, we need the agent to wait until the mount is done, not to write in the wrong spot